azure-pipelines/azure-templates/build-template.yml

```diff
@@ -0,0 +1,51 @@
+
+parameters:
+  - name: 'displayName'
+    type: string
+  - name: 'mavenPomFile'
+    type: string
+  - name: 'options'
+    default: ''
+    type: string
+  - name: 'targetPath'
+    type: string
+  - name: 'artifact'
+    type: string
+
+
+steps:
+- task: SonarCloudPrepare@1
+  inputs:
+    SonarCloud: 'SonarCloud'
+    organization: '1054-Daniel-Felleman-Revature-Swag-Shop'
+    scannerMode: 'Other'
+# - task: Maven@3
+#   inputs:
+#     mavenPomFile: ${{ parameters.mavenPomFile }}
+#     options: ${{ parameters.options }}
+#     publishJUnitResults: true
+#     testResultsFiles: '**/surefire-reports/TEST-*.xml'
+#     codeCoverageToolOption: 'JaCoCo'
+#     codeCoverageRestoreOriginalPomXml: true
+#     javaHomeOption: 'JDKVersion'
+#     jdkVersionOption: '16.0.1'
+#     mavenVersionOption: 'Default'
+#     mavenDirectory: /usr/local/bin/apache-maven-3.8.1/bin
+#     mavenAuthenticateFeed: false
+#     effectivePomSkip: false
+#     sonarQubeRunAnalysis: true
+#     isJacocoCoverageReportXML: true
+- script: |
+    java --version
+    mvn clean package -f ${{ parameters.mavenPomFile }}
+
+- task: SonarCloudPublish@1
+  inputs:
+    pollingTimeoutSec: '300'
+
+- task: PublishPipelineArtifact@1
+  inputs:
+    targetPath: ${{ parameters.targetPath }}
+    artifact: ${{ parameters.artifact }}
+    publishLocation: 'pipeline'
```

Comment:
i don't know why everyone keeps putting publishLocation in when the docs say it's not part of this task, but maybe the docs are just wrong. just double check it

Reply:
Azure Pipelines is putting there.